vms/platformvm/cache_internal_state.go

```diff
@@ -167,12 +178,15 @@ type internalStateImpl struct {
 
 	currentStakerChainState currentStakerChainState
 	pendingStakerChainState pendingStakerChainState
+	lockChainState          lockChainState
 
 	currentHeight         uint64
 	addedCurrentStakers   []*validatorReward
 	deletedCurrentStakers []*Tx
 	addedPendingStakers   []*Tx
 	deletedPendingStakers []*Tx
+	addedLocks            []*validatorReward
+	deletedLocks          []*Tx
```

Comment:
Well, added locks and deleted locks are just temporary things that a cleared after write to db
But yea, whole state thing must be as small as possible, cause it will take ram. There is possible case when we leave some historical things just in db, not in the current state in ram (dao have this, if remember correctly), but it will also take disk space in both cases, yep